ADR/user-specified-tokens-build.md

```diff
@@ -0,0 +1,75 @@
+# Enable users to configure tokens/credentials for a Build/Test Pipeline
+
+Date: 2023-03-03
+
+## Status
+
+Provisional
+
+
+## Context
+
+As a user, I would like to configure sensitive information/tokens/credentials for the build/test pipeline to consume as part of a partner Task.
+
+## What is a Partner Task
+
+A Partner Task is a Tekton Task that is provided and maintained by a Red Hat Partner. However, this initiative also aims to enable Tekton Tasks 
+contributed to the StoneSoup Build/Test Pipeline by different organizations inside Red Hat.
+
+
+## Decision
+
+### How would the sensitive information be persisted and made available to the PipelineRun ?
+
+1. The UI would take in sensitive information as form input and upload it to SPI using the user's authentication token. See https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#uploading-access-token-to-spi-using-kubernetes-secret .
+
+2. Subsequently, the UI would create an `SPIAccessTokenBinding` CR with the appropriate label in the `.spec.secret.labels` to make the `Secrets` available to the build service or the integration service.
+
+
+### How does the build service and the integration service discover the `Secrets` ?
+
+While uploading a token, the user should be able to choose the name of the Pipeline `parameter` from a UI dropdown. The dropdown would be a pre-populated and hard-coded list of names associated with known partner integrations. 
+
+The user should also be allowed to provide a parameter name of choice. The same could be used in a customized PipelineRun definition in the user's source code's .tekton/ directory.
+
+| Pipeline Parameter    | Contents of the Secret | Name of the Secret        |
+| --------------------  | ---------------------- | ------------------------  |
+| codecov-github-token  | *input from the user*  | *autogenerated*           | 
+| codecov-token-secret  | *input from the user*  | *autogenerated*           |
+| redhat-ocm-token      | *input from the user*  | *autogenerated*           | 
```

Comment:
These partner integration should be included in our default build pipelines? Otherwise users would need to define them in .tekton dir which is not user-friendly.